src/app/core/cart/services/cart-api.service.ts

```diff
@@ -0,0 +1,67 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { Observable, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+import { environment } from '../../../../environments/environment';
+import API_ENDPOINT from '../../../shared/constants/api-endpoint';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CartApiService {
+  private readonly CARTS_URL = `${environment.apiUrl}/${environment.projectKey}/${API_ENDPOINT.CARTS}`;
+  private readonly ME_CARTS_URL = `${environment.apiUrl}/${environment.projectKey}/me/${API_ENDPOINT.CARTS}`;
+
+  constructor(private http: HttpClient) {}
+
+  public getCartByCustomerId(customerId: string): Observable<unknown> {
+    return this.http.get(`${this.CARTS_URL}/customer-id=${customerId}`).pipe(
```

Comment:
Avoid using unknown type.

```suggestion
  public getCartByCustomerId(customerId: string): Observable<CartResponse> {
    return this.http.get<CartResponse>(`${this.CARTS_URL}/customer-id=${customerId}`).pipe(
```

Reply:
Done